xlators/protocol/client/src/client-helpers.c

```diff
@@ -417,7 +417,8 @@ client_get_remote_fd(xlator_t *this, fd_t *fd, int flags, int64_t *remote_fd)
             if (fd->anonymous) {
                 *remote_fd = GF_ANON_FD_NO;
             } else {
-                *remote_fd = -1;
+                if (this->fops->writev && conf->strict_locks)
+                    *remote_fd = -EBADFD;
```

Comment:
When the strict-locks is enable we should not set the remote_fd to GF_ANON_FD_NO with write fop. Otherwise even though the fd is bad, the write will succeed on the bad brick. To avoid that I have added this check, so that @436, it won't go and set the remote-fd to GF_ANON_FD_NO.

Reply:
When will this->fops->writev be NULL?

Reply:
Maybe you can rename locks_held to locks_involved and set it to true in both these cases. Code will be a bit simpler.

Reply:
> When will this->fops->writev be NULL?
This may not be NULL for fops other than write, hence removed this.

> Maybe you can rename locks_held to locks_involved and set it to true in both these cases. Code will be a bit simpler.
Done.